app/form_objects/edit_initial_request_form.rb

```diff
@@ -0,0 +1,28 @@
+class EditInitialRequestForm
```

Comment:
Both this form and the [InitialRequest form](https://github.com/DFE-Digital/publish-teacher-training/blob/master/app/form_objects/initial_request_form.rb) have the `number of places` validation in common - potentially I could extract the common behaviour to a base class an inherit but then it may not be obvious what the base class is doing. Any opinion?

Reply:
@asmega @aliuk2012 and I have been looking at this - it would be better if we could have separate form objects for each of these pages - rather than overloading one form object with lots of different validations. We can also use some inbuilt rails validation to more closely match what happens on the API side

`validates :number_of_places, presence: { message: "You must enter a number" },
                               numericality: { only_integer: true, greater_than: 0, message: "You must enter a number greater than 0" }`

Reply:
@aliuk2012 @asmega @SamJamCul - I think using separate form objects is potentially an idea worth considering - there may be some factors to consider though. At the code review we opted for a single for a single form object for all pages to more close mirror the 'create' initial allocation process. If we go for a multiple form object pattern (i.e. one form object per page) we will need to refactor the create initial allocation process too, so will require some more work and will have to make a case for prioritising. Also are suggesting that the multiple form object pattern become the preferred approach moving forwards? Could be worth getting input from the rest of the team.